cmake/public/utils.cmake

```diff
@@ -446,10 +459,18 @@ function(torch_compile_options libname)
 
     if(MSVC)
     elseif(WERROR)
-      target_compile_options(${libname} PRIVATE -Wno-strict-overflow)
+      list(APPEND private_compile_options -Wno-strict-overflow)
     endif()
   endif()
 
+  target_compile_options(${libname} PRIVATE
+      $<$<COMPILE_LANGUAGE:CXX>:${private_compile_options}>)
+  if(USE_CUDA)
+    string(REPLACE ";" "," private_compile_options "${private_compile_options}")
```

Comment:
Can you please add a short comment explaining that this converts list into string. 
And wouldn't `foreach` loop express the intent in a clearer way in this case?

Reply:
I've added a comment to make the intention clear. I really do want a comma-separated list though, so a foreach would just end up as more code IMO.

Reply:
Sounds good, but I wonder if we need to check if private_compile_options is space-free?